source/fundamentals/connections/connection-options.txt

```diff
@@ -0,0 +1,201 @@
+     - string
+     - ``null``
+     - Specifies the replica set name for the cluster. All nodes in the
+       replica set must have the same replica set name, or the Client will not
+       consider them as part of the set.
+
+   * - **maxIdleTimeMS**
+     - integer
+     - ``0``
+     - Specifies the maximum amount of time a connection can remain idle
+       in the connection pool before being removed and closed. The
+       default is ``0``, meaning a connection can remain unused
+       indefinitely.
+
+   * - **minPoolSize**
+     - integer
+     - ``0``
+     - Specifies the minimum number of connections that the driver maintains
+       in a single connection pool.
+
+   * - **serverSelectionTimeoutMS**
+     - integer
+     - ``30000``
+     - Specifies the number of milliseconds to wait to find an available,
+       suitable server to execute an operation.
+
+   * - **heartbeatFrequencyMS**
+     - integer
+     - ``10000``
+     - Specifies the number of milliseconds to wait between periodic
+       background server checks.
+
+   * - **tls**
+     - boolean
+     - ``false``
+     - Specifies whether to establish a Transport Layer Security (TLS)
+       connection with the instance. This is automatically set to ``true``
+       when using a DNS seedlist (SRV) in the connection string. You can
+       override this behavior by setting the value to ``false``.
+
+   * - **w**
+     - string or integer
+     - ``null``
+     - Specifies the write concern. To learn more about values, see the
+       server documentation on
+       :manual:`Write Concern options </reference/write-concern>`.
+
+   * - **directConnection**
+     - boolean
+     - ``false``
+     - Specifies whether to force dispatch **all** operations to the host
+       specified in the connection URI.
+
+For a full list of connection options, see the `ClientOptions API
+documentation
+<{+api+}/mongo/options#ClientOptions>`__.
+
+.. _golang-timeout-setting:
+
+Single Timeout Setting
+----------------------
+
+You can set a single ``Timeout`` option on your ``Client`` instance to
+specify the maximum amount of time that a single operation can take to
+execute.
+
+Set a timeout by calling the ``SetTimeout()`` method when specifying
+options for your ``Client`` instance or by specifying the
+``timeoutMS`` option in your connection URI. By default, all
+``Database``, ``Collection``, ``Session``, ``ChangeStream``, and
+``Bucket`` instances elsewhere in your application inherit the
+``Timeout`` option from ``Client`` if you do not set a different timeout
+on specific operations in the operation's Context.
+
+If you set a timeout on a Context passed into an operation, the driver uses
+that value for the operation. If you do not specify a Context timeout,
+the operation Context derives the timeout from the ``Client`` instance.
+
+.. note:: Retries under Timeout Specification
+
+   When using default settings, if you set a ``Timeout`` option on your ``Client``
```

Comment:
This behavior is true for a client-level or operation-level (context.Context) CSOT. I would also clarify that this only happens if the server returns a "retryable" error, such as a socket exception.

https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.md#retryability